api/src/schedules/retention.ts

```diff
@@ -0,0 +1,127 @@
+import { Action } from '@directus/constants';
+import { useEnv } from '@directus/env';
+import { toBoolean } from '@directus/utils';
+import type { Knex } from 'knex';
+import getDatabase from '../database/index.js';
+import { useLock } from '../lock/index.js';
+import { useLogger } from '../logger/index.js';
+import { getMilliseconds } from '../utils/get-milliseconds.js';
+import { scheduleSynchronizedJob, validateCron } from '../utils/schedule.js';
+
+export interface RetentionTask {
+	collection: string;
+	where?: readonly [string, string, Knex.Value | null];
+	join?: readonly [string, string, string];
+	timeframe: number | undefined;
+}
+
+const env = useEnv();
+
+const retentionLockKey = 'schedule--data-retention';
+const retentionLockTimeout = 10 * 60 * 1000; // 10 mins
+
+const ACTIVITY_RETENTION_TIMEFRAME = getMilliseconds(env['ACTIVITY_RETENTION']);
+const FLOW_LOGS_RETENTION_TIMEFRAME = getMilliseconds(env['FLOW_LOGS_RETENTION']);
+const REVISIONS_RETENTION_TIMEFRAME = getMilliseconds(env['REVISIONS_RETENTION']);
+
+const RETENTION_TASKS: RetentionTask[] = [
```

Comment:
since the value of this variable changes after initialisation it seems not  
appropriate to write it in uppercase

Reply:
Agreed, originally it did not change and is why it is capital.

Reply:
Resolved in 8f24a1698be53ec4dc56f82eb7129fd294f2295f